source/adios2/engine/hdf5/HDF5ReaderP.cpp

```diff
@@ -10,4 +10,299 @@
 
 #include "HDF5ReaderP.h"
 
-#include <iostream> //needs to go away, this is just for demo purposes
+#include "adios2/ADIOSMPI.h"
+
+namespace adios
+{
+
+HDF5Reader::HDF5Reader(ADIOS &adios, const std::string name,
+                       const std::string accessMode, MPI_Comm mpiComm,
+                       const Method &method)
+: Engine(adios, "HDF5Reader", name, accessMode, mpiComm, method,
+         " HDF5Reader constructor (or call to ADIOS Open).\n")
+{
+    Init();
+    MPI_Comm_rank(mpiComm, &_mpi_rank);
+    MPI_Comm_size(mpiComm, &_mpi_size);
+}
+
+HDF5Reader::~HDF5Reader() {}
+
+void HDF5Reader::Init()
+{
+    if (m_AccessMode != "r" && m_AccessMode != "read")
+    {
+        throw std::invalid_argument(
+            "ERROR: HDF5Reader doesn't support access mode " + m_AccessMode +
+            ", in call to ADIOS Open or HDF5Reader constructor\n");
+    }
+
+    _H5File.H5_Init(m_Name, m_MPIComm, false);
+    _H5File.GetNumTimeSteps();
+}
+
+Variable<void> *HDF5Reader::InquireVariable(const std::string &variableName,
+                                            const bool readIn)
+{
+    std::cout << "Not implemented: HDF5Reder::InquireVariable()" << std::endl;
+}
+
+Variable<char> *HDF5Reader::InquireVariableChar(const std::string &variableName,
+                                                const bool readIn)
+{
+}
+
+Variable<unsigned char> *
+HDF5Reader::InquireVariableUChar(const std::string &variableName,
+                                 const bool readIn)
+{
+}
+
+Variable<short> *
+HDF5Reader::InquireVariableShort(const std::string &variableName,
+                                 const bool readIn)
+{
+}
+
+Variable<unsigned short> *
+HDF5Reader::InquireVariableUShort(const std::string &variableName,
+                                  const bool readIn)
+{
+}
+
+Variable<int> *HDF5Reader::InquireVariableInt(const std::string &variableName,
+                                              const bool)
+{
+}
+
+Variable<unsigned int> *
+HDF5Reader::InquireVariableUInt(const std::string &variableName, const bool)
+{
+}
+
+Variable<long int> *
+HDF5Reader::InquireVariableLInt(const std::string &variableName, const bool)
+{
+}
+
+Variable<unsigned long int> *
+HDF5Reader::InquireVariableULInt(const std::string &variableName, const bool)
+{
+}
+
+Variable<long long int> *
+HDF5Reader::InquireVariableLLInt(const std::string &variableName, const bool)
+{
+}
+
+Variable<unsigned long long int> *
+HDF5Reader::InquireVariableULLInt(const std::string &variableName, const bool)
+{
+}
+
+Variable<float> *
+HDF5Reader::InquireVariableFloat(const std::string &variableName, const bool)
+{
+}
+
+Variable<double> *
+HDF5Reader::InquireVariableDouble(const std::string &variableName, const bool)
```

Comment:
I added null return, for now. 

This function has a body now, because I noticed ReadMe(), which is not inherited from any parent class, was not touched during compilation, unless I call it from a parent class function, like InquireVariableDouble().

Do you know why?